src/PrestaShopBundle/Bridge/Helper/HelperListBridge.php

```diff
@@ -0,0 +1,523 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Bridge\Helper;
+
+use Configuration;
+use Context;
+use Db;
+use DbQuery;
+use HelperList;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Bridge\Controller\ControllerConfiguration;
+use PrestaShopBundle\Bridge\Utils\CookieFilterUtils;
+use PrestaShopBundle\Security\Admin\Employee;
+use PrestaShopException;
+use Shop;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorage;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Tools;
+use Validate;
+
+/**
+ * A bridge to use helper list to render list in Controller migrate horizontally
+ */
+class HelperListBridge
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @var Employee
+     */
+    private $user;
+
+    /**
+     * @var HelperListVarsAssigner
+     */
+    private $helperListVarsAssigner;
+
+    /**
+     * @var HookDispatcherInterface
+     */
+    private $hookDispatcher;
+
+    public function __construct(
+        LegacyContext $legacyContext,
+        TokenStorage $tokenStorage,
+        HelperListVarsAssigner $helperListVarsAssigner,
+        HookDispatcherInterface $hookDispatcher
+    ) {
+        $this->context = $legacyContext->getContext();
+        $this->user = $this->getUser($tokenStorage);
+        $this->helperListVarsAssigner = $helperListVarsAssigner;
+        $this->hookDispatcher = $hookDispatcher;
+    }
+
+    /**
+     * Render list content
+     *
+     * @var HelperListConfiguration
+     *
+     * @return string
+     */
+    public function renderList(
+        ControllerConfiguration $controllerConfiguration,
```

Comment:
Why does the list need access to the controller's configuration?. I feel that HelperList and the controller are still too closely coupled. Here you should be sending only HelperListConfiguration